internal/pkg/deploy/env.go

```diff
@@ -18,6 +18,16 @@ const (
 	EnvTemplateVersionBootstrap = "bootstrap"
 )
 
+type stackConfigurer interface {
+	Parameters() (string, error)
```

Comment:
Why is `Parameters` a single `string`? I probably wrote somethign like for the previous addons 😅 feels strange maybe there was a good reason

Reply:
Perhaps we could rename the interface:
```go
type nestedStackConfigurer interface { // Rename the interface because `CustomParameters` is specific to nested stacks.
  Parameters() (string, error)  // Rename the method.
}
```


Basically, this `Parameters()` method is slightly different than the `Parameters()` of other "stack configurer"s - for example `EnvStackConfig` - that returns a list of `cloudformation.Parameter`s. Those `Parameters()` gives values to the "Parameters" section of a stack. They are later fed into an API call. 

This `Parameters()` method is meant to pass values from the parent stack to the addons stack's parameters, and is later rendered into the template.
```yaml
{{- if .Addons}}
  AddonsStack:
    Metadata:
      'aws:copilot:description': 'An addons CloudFormation stack for your additional AWS resources'
    Type: AWS::CloudFormation::Stack
    Properties:
      Parameters:
        App: !Ref AppName
        Env: !Ref EnvironmentName
        {{- if .Addons.ExtraParams }} # <- This is from `Parameters() (string, error)`
        {{ .Addons.ExtraParams }}
        {{- end}}
      TemplateURL: {{.Addons.URL}}
{{- end }}
```

This is due to the different ways of how a stack is configured (via API) vs. how a nested stack is configured (from parent).